jobs/tcp_router/templates/tcp_router.yml.erb

```diff
@@ -91,3 +91,50 @@ backend_tls:
   client_cert_and_key_path: "/var/vcap/jobs/tcp_router/config/keys/tcp-router/backend/client_cert_and_key.pem"
 <% end %>
 <% end -%>
+
+<%
+  frontend_tls = p('tcp_router.frontend_tls', [])
+
+  if frontend_tls&.any?
+    frontend_tls.each_with_index do |cert_pair, i|
+      unless cert_pair.is_a?(Hash) &&
+             cert_pair.key?('cert_chain') &&
+             cert_pair.key?('private_key')
+        raise "frontend_tls[#{i}] must include both cert_chain and private_key"
+      end
+
+      cert_chain = cert_pair['cert_chain'].to_s.strip
+      private_key = cert_pair['private_key'].to_s.strip
+
+      if private_key.empty?
+        raise "frontend_tls[#{i}].private_key must not be empty"
+      end
+
+      cert = OpenSSL::X509::Certificate.new(cert_chain)
+
+      has_san = cert.extensions.any? do |ext|
+        ext.oid == 'subjectAltName' || ext.oid == '2.5.29.17'
+      end
+
+      unless has_san
+        raise "frontend_tls[#{i}].cert_chain must include a subjectAltName extension"
+      end
+    end
+  end
+%>
+
+<%
+def indent(text, spaces)
+  prefix = ' ' * spaces
+  text.to_s.lines.map { |line| prefix + line }.join
+end
+-%>
+
+frontend_tls:
+<% p('tcp_router.frontend_tls', []).each do |pair| -%>
+  - cert_chain: |
+<%= indent(pair['cert_chain'], 6) %>
+    private_key: |
+<%= indent(pair['private_key'], 6) %>
+<% end -%>
```

Comment:
Why are we passing the tls configs as tcp-router config, and also in the pem file listed below? Shouldn't we only need them in one of those two places?